src/algebra/free_monoid.lean

```diff
@@ -28,39 +28,43 @@ def free_monoid (α) := list α
 namespace free_monoid
 
 @[to_additive]
-instance {α} : monoid (free_monoid α) :=
+instance : monoid (free_monoid α) :=
 { one := [],
   mul := λ x y, (x ++ y : list α),
   mul_one := by intros; apply list.append_nil,
   one_mul := by intros; refl,
   mul_assoc := by intros; apply list.append_assoc }
 
 @[to_additive]
-instance {α} : inhabited (free_monoid α) := ⟨1⟩
+instance : inhabited (free_monoid α) := ⟨1⟩
 
 @[to_additive]
-lemma one_def {α} : (1 : free_monoid α) = [] := rfl
+lemma one_def : (1 : free_monoid α) = [] := rfl
 
 @[to_additive]
-lemma mul_def {α} (xs ys : list α) : (xs * ys : free_monoid α) = (xs ++ ys : list α) :=
+lemma mul_def (xs ys : list α) : (xs * ys : free_monoid α) = (xs ++ ys : list α) :=
 rfl
 
 /-- Embeds an element of `α` into `free_monoid α` as a singleton list. -/
 @[to_additive "Embeds an element of `α` into `free_add_monoid α` as a singleton list." ]
 def of (x : α) : free_monoid α := [x]
 
 @[to_additive]
-lemma of_mul_eq_cons (x : α) (l : free_monoid α) : of x * l = x :: l := rfl
+lemma of_def (x : α) : of x = [x] := rfl
+
+@[to_additive]
```

Comment:
```suggestion
/-- Recursor for `free_monoid` using `1` and `of x * xs` instead of `[]` and `x :: xs`. -/
@[to_additive "Recursor for `free_add_monoid` using `0` and `of x + xs` instead of `[]` and `x :: xs`."]
```